### PR TITLE
SDIT-1617 Align JPA booking with SQL results

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/AgencyInternalLocation.java [hunk 1]
```diff
@@ -83,6 +83,10 @@ public class AgencyInternalLocation {
     @Column(name = "AGY_LOC_ID")
     private String agencyId;
 
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "AGY_LOC_ID", nullable = false, updatable = false, insertable = false)
+    private AgencyLocation location;
+
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/AgencyInternalLocation.java:89]: Double map agency so we can get its description

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java [hunk 2]
```diff
@@ -479,7 +482,7 @@ private static LocalDate deriveOffenderReleaseDate(final KeyDateValues keyDateVa
 
 
     public Optional<LocalDate> getConfirmedReleaseDate() {
-        return Optional.ofNullable(releaseDetail != null ? releaseDetail.getReleaseDate() != null ? releaseDetail.getReleaseDate() : releaseDetail.getAutoReleaseDate() : null);
+        return Optional.ofNullable(releaseDetail != null ? releaseDetail.getReleaseDate() : null);
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:485]: The old SQL solution didn't default auto release date into confirmed release date if release date missing

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java [hunk 3]
```diff
@@ -740,6 +743,54 @@ public List<BookingAdjustment> getBookingAdjustments() {
             .toList();
     }
 
+    public String getLocationDescription() {
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:746]: Unpicked the code in InmateService

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderIdentifier.java [hunk 4]
```diff
@@ -62,6 +63,9 @@ public static class OffenderIdentifierPK implements Serializable {
     @Column(name = "CASELOAD_TYPE")
     private String caseloadType;
 
+    @Column(name = "CREATE_DATETIME", insertable = false, updatable = false)
+    private LocalDateTime createDateTime;
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderIdentifier.java:67]: The old SQL solution included the created date for an offender identifier in property `whenCreated` on the API model

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**